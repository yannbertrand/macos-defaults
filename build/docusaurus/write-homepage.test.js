const fs = require('fs')
jest.mock('fs')

const writeHomepage = require('./write-homepage')

const templatesPath = 'templates'
const destinationPath = './dist'

describe('write-homepage', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('no categories', () => {
    beforeEach(() => callWriteHomepage({ categories: null }))

    it('should write an empty docs/readme.md file using the home template', () => {
      const docsReadmeContent = readFile(`${destinationPath}/docs/readme.md`)
      expect(docsReadmeContent).toMatchSnapshot()
    })
  })

  describe('one category, no page', () => {
    beforeEach(() =>
      callWriteHomepage({
        categories: [
          {
            folder: 'category',
            name: 'Category',
            description: 'Category description.',
            image: {
              filename: 'category.png',
              width: 740,
              height: 80
            }
          }
        ]
      })
    )

    it('should write a docs/readme.md file using the home template', () => {
      const docsReadmeContent = readFile(`${destinationPath}/docs/readme.md`)
      expect(docsReadmeContent).toMatchSnapshot()
    })
  })

  describe('one category, one page', () => {
    beforeEach(() =>
      callWriteHomepage({
        categories: [
          {
            folder: 'category',
            name: 'Category',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'string' },
                examples: [
                  {
                    value: '~/Desktop',
                    default: true,
                    text: 'output when value is ~/Desktop'
                  },
                  {
                    value: '~/Pictures',
                    text: 'output when value is ~/Pictures'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          }
        ]
      })
    )

    it('should write a docs/readme.md file using the home template', () => {
      const docsReadmeContent = readFile(
        `${destinationPath}/docs/readme.md`
      )
      expect(docsReadmeContent).toMatchSnapshot()
    })
  })

  describe('one category, two pages', () => {
    beforeEach(() =>
      callWriteHomepage({
        categories: [
          {
            folder: 'category',
            name: 'Category',
            keys: [
              {
                key: 'page1',
                domain: 'com.apple.category',
                title: 'Page 1',
                description: 'Page 1 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              },
              {
                key: 'page2',
                domain: 'com.apple.category',
                title: 'Page 2',
                description: 'Page 2 description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          }
        ]
      })
    )

    it('should write a docs/readme.md file using the home template', () => {
      const docsReadmeContent = readFile(`${destinationPath}/docs/readme.md`)
      expect(docsReadmeContent).toMatchSnapshot()
    })
  })

  describe('two categories, one page in each', () => {
    beforeEach(() =>
      callWriteHomepage({
        categories: [
          {
            folder: 'category1',
            name: 'Category 1',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category1',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    default: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          },
          {
            folder: 'category2',
            name: 'Category 2',
            keys: [
              {
                key: 'page',
                domain: 'com.apple.category2',
                title: 'Page',
                description: 'Page description.',
                param: { type: 'bool' },
                examples: [
                  {
                    value: true,
                    text: 'output when value is true'
                  },
                  {
                    value: false,
                    default: true,
                    text: 'output when value is false'
                  }
                ],
                versions: ['Big Sur']
              }
            ]
          }
        ]
      })
    )

    it('should write a docs/readme.md file using the home template', () => {
      const docsReadmeContent = readFile(`${destinationPath}/docs/readme.md`)
      expect(docsReadmeContent).toMatchSnapshot()
    })
  })
})

const callWriteHomepage = defaults => writeHomepage(defaults, templatesPath, destinationPath)
const readFile = file => fs.readFakeFileSync(file, 'utf8')
