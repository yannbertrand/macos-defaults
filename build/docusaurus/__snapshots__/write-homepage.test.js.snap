// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-homepage no categories should write an empty docs/readme.md file using the home template 1`] = `
"---
id: index
title: Home
sidebar_label: Home
---

Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it using [Docusaurus](https://docusaurus.io/). Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/docusaurus#readme) if you want to use it.
"
`;

exports[`write-homepage one category, no page should write a docs/readme.md file using the home template 1`] = `
"---
id: index
title: Home
sidebar_label: Home
---

Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## Category

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it using [Docusaurus](https://docusaurus.io/). Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/docusaurus#readme) if you want to use it.
"
`;

exports[`write-homepage one category, one page should write a docs/readme.md file using the home template 1`] = `
"---
id: index
title: Home
sidebar_label: Home
---

Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## Category
- [Page](category/page)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it using [Docusaurus](https://docusaurus.io/). Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/docusaurus#readme) if you want to use it.
"
`;

exports[`write-homepage one category, two pages should write a docs/readme.md file using the home template 1`] = `
"---
id: index
title: Home
sidebar_label: Home
---

Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## Category
- [Page 1](category/page1)
- [Page 2](category/page2)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it using [Docusaurus](https://docusaurus.io/). Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/docusaurus#readme) if you want to use it.
"
`;

exports[`write-homepage two categories, one page in each should write a docs/readme.md file using the home template 1`] = `
"---
id: index
title: Home
sidebar_label: Home
---

Uncomplete list of macOS \`defaults\` commands with demos ✨

## 💻 Commands
## Category 1
- [Page](category1/page)

## Category 2
- [Page](category2/page)

## 🤔 How do I add a command?
Please update [this file](https://github.com/yannbertrand/macos-defaults/blob/master/defaults.yml) that I use to [build multiple websites](https://github.com/yannbertrand/macos-defaults/#readme) like this one.

## ❤️ I like this website, how can I build the same?
Thank you! I built it using [Docusaurus](https://docusaurus.io/). Take a look at [my report](https://github.com/yannbertrand/macos-defaults/tree/master/build/docusaurus#readme) if you want to use it.
"
`;
